sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp

```diff
@@ -378,6 +432,27 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     PathAccessConditions AccessConditions;
   };
 
+  using ScheduleFileExpiryOriginType = Blobs::Models::ScheduleBlobExpiryOriginType;
+
+  /**
+   * @brief Optional parameters for #Azure::Storage::Files::DataLake::FileClient::UploadFrom.
+   */
+  struct ScheduleFileDeletionOptions final
+  {
+    /**
+     * The expiry time from the specified origin. Only work if ExpiryOrigin is
+     * ScheduleFileExpiryOriginType::RelativeToCreation or
+     * ScheduleFileExpiryOriginType::RelativeToNow.
+     */
+    Azure::Nullable<std::chrono::milliseconds> TimeToExpire;
```

Comment:
What is the difference between `TimeToExpire` and `ExpiresOn`? 

And why is `TimeToExpire` expressed in milliseconds? That's a surprising unit of time (if it's a relative time, I would expect it to be something closer to std::chrono::duration).

Reply:
There are four types in ScheduleFileExpiryOriginType : 
<html>
<body>
<!--StartFragment-->

Expiry Option | Description
-- | --
RelativeToCreation | Sets the expiry time relative to the file creation time, x-ms-expiry-time must be specified as the number of milliseconds to elapse from creation time.
RelativeToNow | Sets the expiry relative to the current time, x-ms-expiry-time must be specified as the number of milliseconds to elapse from now.
Absolute | x-ms-expiry-time must be specified as an absolute time in RFC 1123 Format.
NeverExpire | Sets the file to never expire or removes the current expiry time, x-ms-expiry-time must not to be specified.

<!--EndFragment-->
</body>
</html>

And then,
**TimeToExpire** is a relative time to Expire from now/creationTime in miliseconds, It will work under **RelativeToCreation** and **RelativeToNow**  types. Since these types are specified as the number of milliseconds, I think using milliseconds is esier for users to understand the unit of time supported by the REST API.
**ExpiresOn** is a absolute time(RFC 1123) to Expire. It will work under **Absolute** type.